tests/test_browser.go

```diff
@@ -105,10 +105,14 @@ func newTestBrowser(tb testing.TB, opts ...any) *testBrowser {
 		state      = vu.StateField
 		rt         = vu.RuntimeField
 		lc         *logCache
+		err        error
 	)
 
 	if enableLogCache {
-		lc = attachLogCache(state.Logger)
+		lc, err = attachLogCache(state.Logger)
+		if err != nil {
+			panic(fmt.Errorf("testBrowser: attaching logger: %w", err))
+		}
```

Comment:
Instead of panicking here, it would make better sense to `t.Fail` like:

```go
attachLogCache(t, state.Logger)
// attachLogCache can now use `tb.Fatalf("unexpected logger type: %T", fl)`
```

The benefit is we won't get a huge panic output because of a few failing tests. And we can continue running the other tests even if those fail. And we don't need to use an error-handling code, as the testing package will handle it for us.

`Fatalf` will print out something like: `--- FAIL: TestBrowserLogIterationID: logrus_hook.go:81: unexpected logger type: *tests.MyLogger`.

Reply:
I used `panic` to follow the pattern applied in the rest of the `testBrowser` implementation (e.g.: [here](https://github.com/grafana/xk6-browser/blob/d506083ca71d697a3116c569133cfc603dd6c5ff/tests/test_browser.go#L99) and [here](https://github.com/grafana/xk6-browser/blob/d506083ca71d697a3116c569133cfc603dd6c5ff/tests/test_browser.go#L126)), but I personally prefer to use `tb.Fatalf` also if we agree on changing all `panic` occurrences. What do you think?

Reply:
Got it 👍 I prefer `Fatal` in the rest too. At least, we can use `Fatal` in this PR. It's fine if you don't want to tackle the rest now.

Reply:
Got it 👍 At least we can use `Fatal` in this PR. It's fine if you don't want to tackle the rest now (as they're unrelated to this PR).

Reply:
I prefer to fix the three `panic` usages to keep it consistent. Is it ok if we do that in this PR?

Reply:
Done in [68a366e](https://github.com/grafana/xk6-browser/pull/842/commits/68a366e77e5ea85d40422b82350a2158d1bb3ff2) and [8740dd4](https://github.com/grafana/xk6-browser/pull/842/commits/8740dd463833653966bfd4da875b7f9c90dbafcb).